drift-client/src/test/java/io/airlift/drift/client/TestDriftMethodInvocation.java

```diff
@@ -0,0 +1,745 @@
+                        .setMaxBackoffDelay(new Duration(1, DAYS))
+                        .setBackoffScaleFactor(2.0),
+                new TestingExceptionClassifier());
+
+        TestingMethodInvocationStat stat = new TestingMethodInvocationStat();
+        AtomicInteger attempts = new AtomicInteger();
+        MockMethodInvoker invoker = new MockMethodInvoker(() -> {
+            int currentAttempts = attempts.getAndIncrement();
+            if (currentAttempts < expectedRetries) {
+                return immediateFailedFuture(createClassifiedException(true, NORMAL, wrapWithApplicationException));
+            }
+            return immediateFuture(SUCCESS);
+        });
+        DriftMethodInvocation methodInvocation = createDriftMethodInvocation(retryPolicy, stat, invoker, new TestingAddressSelector(100), Ticker.systemTicker());
+
+        assertEquals(methodInvocation.get(), SUCCESS);
+        assertEquals(attempts.get(), expectedRetries + 1);
+        stat.assertSuccess(expectedRetries);
+        assertDelays(invoker, retryPolicy, expectedRetries);
+    }
+
+    @Test(timeOut = 60000)
+    public void testBasicRetriesToFailure()
+            throws Exception
+    {
+        testBasicRetriesToFailure(0, true);
+        testBasicRetriesToFailure(1, true);
+        testBasicRetriesToFailure(5, true);
+        testBasicRetriesToFailure(10, true);
+        testBasicRetriesToFailure(0, false);
+        testBasicRetriesToFailure(1, false);
+        testBasicRetriesToFailure(5, false);
+        testBasicRetriesToFailure(10, false);
+    }
+
+    private static void testBasicRetriesToFailure(int expectedRetries, boolean wrapWithApplicationException)
+            throws Exception
+    {
+        RetryPolicy retryPolicy = new RetryPolicy(
+                new DriftClientConfig()
+                        .setMaxRetries(expectedRetries + 10)
+                        .setMinBackoffDelay(new Duration(1, SECONDS))
+                        .setMaxBackoffDelay(new Duration(100, SECONDS))
+                        .setBackoffScaleFactor(2.0),
+                new TestingExceptionClassifier());
+
+        TestingMethodInvocationStat stat = new TestingMethodInvocationStat();
+        AtomicInteger attempts = new AtomicInteger();
+        MockMethodInvoker invoker = new MockMethodInvoker(() -> {
+            int currentAttempts = attempts.getAndIncrement();
+            if (currentAttempts < expectedRetries) {
+                return immediateFailedFuture(createClassifiedException(true, NORMAL, wrapWithApplicationException));
+            }
+            return immediateFailedFuture(createClassifiedException(false, NORMAL, wrapWithApplicationException));
+        });
+        DriftMethodInvocation methodInvocation = createDriftMethodInvocation(
+                retryPolicy,
+                stat,
+                invoker,
+                new TestingAddressSelector(100), Ticker.systemTicker());
```

Comment:
Missing newline for argument. Looks like this was added by IntelliJ in several places